src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -132,14 +138,16 @@ func (c *controller) WaitAndUpdateNodesStatus(status *ControllerStatus) {
 			break
 		}
 		//otherwise, update the progress status and keep waiting
-		log.Infof("Searching for host to change status, number to find %d", len(assistedInstallerNodesMap))
+		log.Infof("Checking if cluster nodes are ready. %d nodes remaining", len(assistedInstallerNodesMap))
 		nodes, err := c.kc.ListNodes()
 		if err != nil {
+			log.WithError(err).Error("Failed to get list of nodes from k8s client")
 			continue
 		}
 		for _, node := range nodes.Items {
 			host, ok := assistedInstallerNodesMap[strings.ToLower(node.Name)]
 			if !ok {
+				log.Warnf("Node %s is not in inventory hosts", strings.ToLower(node.Name))
```

Comment:
it is wrong way to do it. We ignore all the nodes in installed status so they will not be in a list.
Please run this flow and verify logs before pushing